api/create.go

```diff
@@ -79,26 +80,37 @@ func CreateService(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	authHeader := req.Header.Get("Authorization")
-	userId, err := config.ValidateToken(authHeader)
+	apiKeyHeader := req.Header.Get("x-api-key")
+
+	userId, err := config.ValidateUser(authHeader, apiKeyHeader)
```

Comment:
```suggestion
        var err error
        s.UserID, err = config.ValidateUser(authHeader, apiKeyHeader)
```

Reply:
good catch.

Reply:
https://github.com/skant7/spinup/blob/d606c9de145f4d482fdd04110cddbb1435ea56cd/api/create.go#L108
This  won't work then ......actually `/createservice` does not expect a username if using the simple auth and manually assigns the username as "testuser" in the request body struct ,otherwise `ValidateUser()` returns a userID using the jwt and checks it with that in the request body

Reply:
I see. that makes sense.